Examples/Scripts/Python/material_mapping_core.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+
+import acts
+from acts import (
+    MaterialMapper,
+    IntersectionMaterialAssigner,
+    BinnedSurfaceMaterialAccumulater,
+    MaterialMapJsonConverter,
+)
+
+from acts.examples import (
+    Sequencer,
+    WhiteBoard,
+    AlgorithmContext,
+    ProcessCode,
+    RootMaterialTrackReader,
+    RootMaterialTrackWriter,
+    CoreMaterialMapping,
+    JsonMaterialWriter,
+    RootMaterialWriter,
+    JsonFormat,
+)
+
+from acts.examples.dd4hep import (
+    DD4hepDetector,
+    DD4hepDetectorOptions,
+    DD4hepGeometryService,
+)
+
+from acts.examples.odd import getOpenDataDetector, getOpenDataDetectorDirectory
+
+
+def runMaterialMapping(surfaces, inputFile, outputFile, outputMap, loglevel):
+    # Create a sequencer
+    print("Creating the sequencer with 1 thread (inter event information needed)")
+
+    s = Sequencer(numThreads=1)
+
+    # IO for material tracks reading
+    wb = WhiteBoard(acts.logging.INFO)
+
+    # Read material step information from a ROOT TTRee
+    s.addReader(
+        RootMaterialTrackReader(
+            level=acts.logging.INFO,
+            outputMaterialTracks="material-tracks",
+            fileList=[inputFile],
+            readCachedSurfaceInformation=False,
+        )
+    )
+
+    # Assignment setup : Intersection assigner
+    materialAssingerConfig = IntersectionMaterialAssigner.Config()
+    materialAssingerConfig.surfaces = surfaces
+    materialAssinger = IntersectionMaterialAssigner(materialAssingerConfig, loglevel)
+
+    # Accumulation setup : Binned surface material accumulater
+    materialAccumulaterConfig = BinnedSurfaceMaterialAccumulater.Config()
+    materialAccumulaterConfig.materialSurfaces = surfaces
+    materialAccumulater = BinnedSurfaceMaterialAccumulater(
+        materialAccumulaterConfig, loglevel
+    )
```

Comment:
As I was reading this I realised something (I think). With your new setup we don't have a different mapping type for homogenous surface any more I think ? Maybe something to add to the todo list ?

Reply:
Homogneous will just be a 1x1 binning, and then turned into a homogenous material when the maps are finalised.